### ISSUE TITLE
dotd/dotc do crash when an option value of type String is missing

### ISSUE THREAD

**Issue by michelou** [lampepfl/dotty#8124]
## Minimized code

```
> C:\opt\dotty-0.22.0-RC1\bin\dotd -project -project-url "http://" dummy.scala
```
__Description__:
1. Value for option `-project` __*is missing*__ (e.g. undefined environment variable in my case).
2. Value for option `-project-url` starts with a valid URL protocol (actual URL is not relevant).
3. Input file `dummy.scala` must not exist.

__Environment__:
- Win10 (Java 8 and 11)

__Reminder__:
```
> c:\opt\dotty-0.22.0-RC1\bin\dotd --help 2>&1 | findstr project
-project           The name of the project
-project-logo      The file that contains the project's logo (in /images)
-project-url       The source repository of your project
-project-version   The current version of your project
Site project name not set. Use `-project <title>` to set the project name
```

## Crash output (click arrow to expand)
<details>

```
java.lang.NullPointerException while compiling http://, dummy.scala
Exception in thread "main" java.lang.NullPointerException
        at dotty.tools.dotc.printing.Formatting$StringFormatter.showArg(Formatting.scala:37)
        at dotty.tools.dotc.printing.Formatting$StringFormatter.treatArg(Formatting.scala:47)
        at dotty.tools.dotc.printing.Formatting$StringFormatter.$anonfun$2(Formatting.scala:60)
        at scala.collection.LazyZip2$$anon$1$$anon$2.next(LazyZipOps.scala:42)
        at scala.collection.mutable.Growable.addAll(Growable.scala:62)
        at scala.collection.mutable.Growable.addAll$(Growable.scala:59)
        at scala.collection.mutable.ArrayBuilder.addAll(ArrayBuilder.scala:69)
        at scala.collection.IterableOnceOps.toArray(IterableOnce.scala:1265)
        at scala.collection.IterableOnceOps.toArray$(IterableOnce.scala:1259)
        at scala.collection.AbstractIterable.toArray(Iterable.scala:921)
        at scala.Array$.from(Array.scala:59)
        at scala.collection.immutable.ArraySeq$.from(ArraySeq.scala:188)
        at scala.collection.immutable.ArraySeq$.from(ArraySeq.scala:179)
        at scala.collection.ClassTagIterableFactory$AnyIterableDelegate.from(Factory.scala:656)
        at scala.collection.BuildFromLowPriority2$$anon$11.fromSpecific(BuildFrom.scala:113)
        at scala.collection.BuildFromLowPriority2$$anon$11.fromSpecific(BuildFrom.scala:110)
        at scala.collection.LazyZip2.map(LazyZipOps.scala:37)
        at dotty.tools.dotc.printing.Formatting$StringFormatter.assemble(Formatting.scala:60)
        at dotty.tools.dotc.core.Decorators$StringInterpolators$.i$extension(Decorators.scala:212)
        at dotty.tools.dotc.Run.compile$$anonfun$1(Run.scala:110)
        at dotty.tools.dotc.reporting.Reporting.echo$$anonfun$1(Reporter.scala:76)
        at dotty.tools.dotc.reporting.diagnostic.MessageContainer.contained(MessageContainer.scala:59)
        at dotty.tools.dotc.reporting.diagnostic.MessageContainer.message(MessageContainer.scala:42)
        at dotty.tools.dotc.reporting.diagnostic.MessageContainer.isNonSensical(MessageContainer.scala:71)
        at dotty.tools.dotc.reporting.HideNonSensicalMessages.isHidden(HideNonSensicalMessages.scala:17)
        at dotty.tools.dotc.reporting.AbstractReporter.isHidden(AbstractReporter.scala:8)
        at dotty.tools.dotc.reporting.Reporter.report(Reporter.scala:270)
        at dotty.tools.dotc.reporting.Reporting.echo(Reporter.scala:76)
        at dotty.tools.dotc.core.Contexts$Context.echo(Contexts.scala:75)
        at dotty.tools.dotc.Run.compile(Run.scala:110)
        at dotty.tools.dotc.Driver.doCompile(Driver.scala:36)
        at dotty.tools.dottydoc.DocDriver.process(DocDriver.scala:39)
        at dotty.tools.dotc.Driver.process(Driver.scala:158)
        at dotty.tools.dotc.Driver.process(Driver.scala:170)
        at dotty.tools.dotc.Driver.main(Driver.scala:197)
        at dotty.tools.dottydoc.Main.main(Main.scala)
```
</details>

**Comment by michelou** [comment]
@liufengyun @smarter Can you please change the labels ?!

**Comment by smarter** [comment]
I've added you as a collaborator to the repo so you should be able to set the labels yourself now :).

**Comment by michelou** [comment]
@smarter Done. Thanx!

**Comment by michelou** [comment]
The same error occurs with `dotc` and a missing value for option `-encoding`.

The handling of `StringSetting` arguments (common to options `-project` and `-encoding`)  in function [`doSet`](https://github.com/lampepfl/dotty/blob/5674066132dda92141b01ad9f029abb069399fe1/compiler/src/dotty/tools/dotc/config/Settings.scala#L130) looks spurious (work in progress).

```
$ c:\opt\dotty-0.22.0-RC1\bin\dotc -encoding -d c:\temp\classes dummy.scala
java.lang.NullPointerException while compiling c:\temp\classes, dummy.scala
Exception in thread "main" java.lang.NullPointerException
        at dotty.tools.dotc.printing.Formatting$StringFormatter.showArg(Formatting.scala:37)
        at dotty.tools.dotc.printing.Formatting$StringFormatter.treatArg(Formatting.scala:47)
        at dotty.tools.dotc.printing.Formatting$StringFormatter.$anonfun$2(Formatting.scala:60)
        at scala.collection.LazyZip2$$anon$1$$anon$2.next(LazyZipOps.scala:42)
        at scala.collection.mutable.Growable.addAll(Growable.scala:62)
        at scala.collection.mutable.Growable.addAll$(Growable.scala:59)
        at scala.collection.mutable.ArrayBuilder.addAll(ArrayBuilder.scala:69)
        at scala.collection.IterableOnceOps.toArray(IterableOnce.scala:1265)
        at scala.collection.IterableOnceOps.toArray$(IterableOnce.scala:1259)
        at scala.collection.AbstractIterable.toArray(Iterable.scala:921)
        at scala.Array$.from(Array.scala:59)
        at scala.collection.immutable.ArraySeq$.from(ArraySeq.scala:188)
        at scala.collection.immutable.ArraySeq$.from(ArraySeq.scala:179)
        at scala.collection.ClassTagIterableFactory$AnyIterableDelegate.from(Factory.scala:656)
        at scala.collection.BuildFromLowPriority2$$anon$11.fromSpecific(BuildFrom.scala:113)
        at scala.collection.BuildFromLowPriority2$$anon$11.fromSpecific(BuildFrom.scala:110)
        at scala.collection.LazyZip2.map(LazyZipOps.scala:37)
        at dotty.tools.dotc.printing.Formatting$StringFormatter.assemble(Formatting.scala:60)
        at dotty.tools.dotc.core.Decorators$StringInterpolators$.i$extension(Decorators.scala:212)
        at dotty.tools.dotc.Run.compile$$anonfun$1(Run.scala:116)
        at dotty.tools.dotc.reporting.Reporting.echo$$anonfun$1(Reporter.scala:77)
        at dotty.tools.dotc.reporting.diagnostic.MessageContainer.contained(MessageContainer.scala:59)
        at dotty.tools.dotc.reporting.diagnostic.MessageContainer.message(MessageContainer.scala:42)
        at dotty.tools.dotc.reporting.diagnostic.MessageContainer.isNonSensical(MessageContainer.scala:71)
        at dotty.tools.dotc.reporting.HideNonSensicalMessages.isHidden(HideNonSensicalMessages.scala:17)
        at dotty.tools.dotc.reporting.AbstractReporter.isHidden(AbstractReporter.scala:8)
        at dotty.tools.dotc.reporting.Reporter.report(Reporter.scala:271)
        at dotty.tools.dotc.reporting.Reporting.echo(Reporter.scala:77)
        at dotty.tools.dotc.core.Contexts$Context.echo(Contexts.scala:75)
        at dotty.tools.dotc.Run.compile(Run.scala:116)
        at dotty.tools.dotc.Driver.doCompile(Driver.scala:36)
        at dotty.tools.dotc.Driver.process(Driver.scala:189)
        at dotty.tools.dotc.Driver.process(Driver.scala:158)
        at dotty.tools.dotc.Driver.process(Driver.scala:170)
        at dotty.tools.dotc.Driver.main(Driver.scala:197)
        at dotty.tools.dotc.Main.main(Main.scala)
```

**Comment by michelou** [comment]
Fixed by pull #8273
